### DISCUSSION TITLE
Gzip content encoding as response from authenticator

### DISCUSSION THREAD

**Discussion by schreddies** [ory/oathkeeper#834] [Ideas]
Right now, the oahtkeeper is not able to work with responses with `Content-Encoding` set as `gzip`. The idea is to add functionality into aothkeeper to be able to work with gziped responses. Pseudocode:
```
if res.Header.Get("Content-Encoding")=="gzip" {
    reader = gzip.NewReader(res.Body)
    body := ioutil.ReadAll(reader)
    return body
}
```
Link to problem, from slack: https://ory-community.slack.com/archives/C01340V8KSM/p1632213654049300

**Comment by danrods** [comment]
Until this has been added is it possible to add some kind of temporary warning message in the authenticator or in the docs, noting that "Gzip is not currently supported"? That way at least theres some kind of traceability

**Reply by vinckr** [reply]
In general, the documentation describes what is possible, not what is not possible/supported. 
But if this particular is causing confusion we can add a short notice at an appropriate section, sure. 
If you already have an idea [where](https://www.ory.sh/oathkeeper/docs/next/pipeline/authz) this could fit in, feel free to open a PR 🤗

**Reply by danrods** [reply]
https://github.com/ory/oathkeeper/pull/835
All set

**Comment by aeneasr** [comment]
Not sure if this is true: https://github.com/ory/oathkeeper/pull/835#pullrequestreview-763578286
